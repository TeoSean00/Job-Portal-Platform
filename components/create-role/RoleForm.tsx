"use client";

import { zodResolver } from "@hookform/resolvers/zod";
import { Controller, useFieldArray, useForm } from "react-hook-form";
import * as z from "zod";

import {
  Button,
  Combobox,
  Input,
  SelectComponent,
  Textarea,
  toast,
  ToastAction,
} from "@/components/ui";
import DatePickerWithRange from "@/components/ui/datePickerWithRange";
import {
  Form,
  FormControl,
  FormDescription,
  FormField,
  FormItem,
  FormLabel,
  FormMessage,
} from "@/components/ui/form";
import { longDateTime } from "@/lib/utils";

type RoleFormProps = {
  skillList: Skill[];
  departments: Department[];
};

const departmentPlaceholder = "Select a department";

const skillSchema = z.object({
  value: z.string().min(1),
});

const roleFormSchema = z.object({
  roleName: z.string().min(1, "Role Name is required").max(100, {
    message: "Role Name must not be longer than 100 characters.",
  }),
  roleDescription: z.string().min(1, "Role Description is required").max(1000),
  departments: z.string(),
  skills: z.array(skillSchema).nonempty("At least one skill is required."),
  dateRange: z.object({
    from: z.date(),
    to: z.date(),
  }),
});

type RoleFormValues = z.infer<typeof roleFormSchema>;

// This can come from your database or API.
const defaultValues: Partial<RoleFormValues> = {
  roleDescription: "",
};

const currentDate = new Date();
const formattedDate = longDateTime.format(currentDate);

const RoleForm: React.FC<RoleFormProps> = ({ skillList, departments }) => {
  const form = useForm<RoleFormValues>({
    resolver: zodResolver(roleFormSchema),
    defaultValues,
    mode: "onChange",
  });

  const { fields, append } = useFieldArray({
    name: "skills",
    control: form.control,
  });

  function onSubmit(data: RoleFormValues) {
    toast({
      title: "Role Successfully Created!",
      description: formattedDate,
      action: <ToastAction altText="ok">Dismiss</ToastAction>,
    });
  }

  // For connecting BE to FE to see the form object put this code into description
  // <pre className="mt-2 flex w-[340px] flex-wrap rounded-md bg-green-950 p-4">
  //         <code className="text-white">{JSON.stringify(data, null, 2)}</code>
  //       </pre>

  return (
    <div className="">
      <Form {...form}>
        <form className="space-y-8" onSubmit={form.handleSubmit(onSubmit)}>
          <FormField
            control={form.control}
            name="roleName"
            render={({ field }) => (
              <FormItem>
                <FormLabel className="text-base">Role Name</FormLabel>
                <FormControl>
                  <Input placeholder="" {...field} />
                </FormControl>
                <FormDescription></FormDescription>
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={form.control}
            name="roleDescription"
            render={({ field }) => (
              <FormItem>
                <FormLabel className="text-base">Role Description</FormLabel>
                <FormDescription>Describe the role in detail.</FormDescription>
                <FormControl>
                  <Textarea className=" resize-y" placeholder="" {...field} />
                </FormControl>
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={form.control}
            name="departments"
            render={({ field: departmentField }) => (
              <FormItem>
                <FormLabel className="text-base">Department</FormLabel>
                <FormDescription>
                  Select the department the role belongs in.
                </FormDescription>
                <Controller
                  control={form.control}
                  name="departments"
                  render={({ field }) => (
                    <Combobox
                      items={departments}
                      placeholder={departmentPlaceholder}
                      value={field.value}
                      onChange={field.onChange}
                    />
                  )}
                />
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={form.control}
            name="skills"
            render={({ field }) => (
              <FormItem>
                <FormLabel className="text-base">Skills</FormLabel>
                <FormDescription>Skills required for role.</FormDescription>
                <SelectComponent
                  createAble={true}
                  isMulti={true}
                  options={skillList}
                  placeholder="Select Skills"
                  {...field}
                />
                <FormMessage />
              </FormItem>
            )}
          />

          <FormField
            control={form.control}
            name="dateRange"
            render={({ field, fieldState }) => (
              <FormItem>
                <FormLabel className="text-base">Date Range</FormLabel>
                <FormDescription>
                  Select the start and end date for this role.
                </FormDescription>
                <Controller
                  control={form.control}
                  name="dateRange"
                  render={({ field: { onChange, value } }) => (
                    <DatePickerWithRange
                      className="w-full"
                      value={value}
                      onChange={(selectedDateRange) => {
                        if (selectedDateRange) {
                          if (selectedDateRange.from && selectedDateRange.to) {
                            onChange({
                              from: selectedDateRange.from,
                              to: selectedDateRange.to,
                            });
                            form.clearErrors("dateRange");
                          } else {
                            form.setError("dateRange", {
                              type: "manual",
                              message:
                                "Both start and end dates must be selected",
                            });
                          }
                        } else {
                          onChange(undefined);
                        }
                      }}
                    />
                  )}
                />
                <FormMessage>
                  {fieldState.error && fieldState.error.message}
                </FormMessage>
              </FormItem>
            )}
          />

          <Button type="submit">Create Role</Button>
        </form>
      </Form>
    </div>
  );
};

export default RoleForm;
