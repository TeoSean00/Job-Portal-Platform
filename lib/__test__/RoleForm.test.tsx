import { render } from "@testing-library/react";
import React from "react";

// import "@testing-library/jest-dom/extend-expect";
import RoleForm from "../../components/create-role/RoleForm";

const mockSkillList = [
  { value: "Pascal Programming", label: "Pascal Programming" },
  { value: "Python Programming", label: "Python Programming" },
  { value: "Certified Scrum Master", label: "Certified Scrum Master" },
  { value: "Marketing", label: "Marketing" },
  { value: "Scrum", label: "Scrum" },
  { value: "UI/UX", label: "UI/UX" },
  { value: "Product Management", label: "Product Management" },
];

const mockDepartments = ["HR", "IT", "Sales", "Finance"].map((department) => ({
  value: department,
  label: department,
}));

describe("<RoleForm />", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <RoleForm departments={mockDepartments} skillList={mockSkillList} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
