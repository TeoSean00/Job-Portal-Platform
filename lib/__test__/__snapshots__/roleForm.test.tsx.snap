// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoleForm /> matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="space-y-8"
  >
    <div
      class="flex w-full flex-grow space-x-4"
    >
      <div
        class="space-y-2 w-full"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
          for=":rd:-form-item"
        >
          Role
        </label>
        <button
          aria-controls="radix-:re:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full justify-between font-normal text-muted-foreground"
          data-state="closed"
          role="combobox"
          type="button"
        >
          Select a Role
          <svg
            class="ml-2 h-4 w-4 shrink-0 opacity-50"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="space-y-2 w-full"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
          for=":rf:-form-item"
        >
          Listing Id
        </label>
        <input
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-4 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
          name="listingId"
          placeholder="id"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      class="space-y-2"
    >
      <label
        class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
        for=":rg:-form-item"
      >
        Listing Description
      </label>
      <p
        class="text-[0.8rem] text-muted-foreground"
        id=":rg:-form-item-description"
      >
        Describe the role in detail.
      </p>
      <textarea
        aria-describedby=":rg:-form-item-description"
        aria-invalid="false"
        class="flex min-h-[60px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 resize-y"
        id=":rg:-form-item"
        name="roleDescription"
        placeholder=""
      />
    </div>
    <div
      class="space-y-2 w-full"
    >
      <label
        class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
        for=":rh:-form-item"
      >
        Role's Manager
      </label>
      <p
        class="text-[0.8rem] text-muted-foreground"
        id=":rh:-form-item-description"
      >
        Select role's manager.
      </p>
      <button
        aria-controls="radix-:ri:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full justify-between font-normal text-muted-foreground"
        data-state="closed"
        role="combobox"
        type="button"
      >
        Select a Manager
        <svg
          class="ml-2 h-4 w-4 shrink-0 opacity-50"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex w-full flex-grow space-x-4"
    >
      <div
        class="space-y-2 w-full"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
          for=":rj:-form-item"
        >
          Location
        </label>
        <p
          class="text-[0.8rem] text-muted-foreground"
          id=":rj:-form-item-description"
        >
          Select the location of role.
        </p>
        <button
          aria-controls="radix-:rk:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full justify-between font-normal text-muted-foreground"
          data-state="closed"
          role="combobox"
          type="button"
        >
          Select location
          <svg
            class="ml-2 h-4 w-4 shrink-0 opacity-50"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="space-y-2 w-full"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
          for=":rl:-form-item"
        >
          Department
        </label>
        <p
          class="text-[0.8rem] text-muted-foreground"
          id=":rl:-form-item-description"
        >
          Select the department the role.
        </p>
        <button
          aria-controls="radix-:rm:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full justify-between font-normal text-muted-foreground"
          data-state="closed"
          role="combobox"
          type="button"
        >
          Select department
          <svg
            class="ml-2 h-4 w-4 shrink-0 opacity-50"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="space-y-2"
    >
      <label
        class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
        for=":rn:-form-item"
      >
        Skills
      </label>
      <p
        class="text-[0.8rem] text-muted-foreground"
        id=":rn:-form-item-description"
      >
        Skills required for role.
      </p>
      <div
        class=" css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="border-border border border-border rounded-lg bg-background hover:cursor-pointer hover:bg-secondary h-10 css-cp01gg-control"
        >
          <div
            class="text-foreground text-sm ml-4 css-1qpkzkj-ValueContainer"
            style="display: grid;"
          >
            <div
              class="text-muted-foreground text-sm ml-1 css-1vlsb4t-placeholder"
              id="react-select-3-placeholder"
            >
              Select Skills
            </div>
            <div
              class="text-foreground text-sm ml-1 css-1q5atpo-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-3-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="p-1 gap-1 bg-background rounded-lg css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="bg-mutated css-j4w2j1-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="p-1 hover:text-foreground text-gray-500 css-g56vrd-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <input
            name="skills"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="space-y-2"
    >
      <label
        class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
        for=":ro:-form-item"
      >
        Start Date
      </label>
      <p
        class="text-[0.8rem] text-muted-foreground"
        id=":ro:-form-item-description"
      >
        Select the date for this listing to be shown.
      </p>
      <div
        class="w-full"
      >
        <button
          aria-controls="radix-:rp:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-[240px] justify-start text-left font-normal text-muted-foreground"
          data-state="closed"
          type="button"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.5 1C4.77614 1 5 1.22386 5 1.5V2H10V1.5C10 1.22386 10.2239 1 10.5 1C10.7761 1 11 1.22386 11 1.5V2H12.5C13.3284 2 14 2.67157 14 3.5V12.5C14 13.3284 13.3284 14 12.5 14H2.5C1.67157 14 1 13.3284 1 12.5V3.5C1 2.67157 1.67157 2 2.5 2H4V1.5C4 1.22386 4.22386 1 4.5 1ZM10 3V3.5C10 3.77614 10.2239 4 10.5 4C10.7761 4 11 3.77614 11 3.5V3H12.5C12.7761 3 13 3.22386 13 3.5V5H2V3.5C2 3.22386 2.22386 3 2.5 3H4V3.5C4 3.77614 4.22386 4 4.5 4C4.77614 4 5 3.77614 5 3.5V3H10ZM2 6V12.5C2 12.7761 2.22386 13 2.5 13H12.5C12.7761 13 13 12.7761 13 12.5V6H2ZM7 7.5C7 7.22386 7.22386 7 7.5 7C7.77614 7 8 7.22386 8 7.5C8 7.77614 7.77614 8 7.5 8C7.22386 8 7 7.77614 7 7.5ZM9.5 7C9.22386 7 9 7.22386 9 7.5C9 7.77614 9.22386 8 9.5 8C9.77614 8 10 7.77614 10 7.5C10 7.22386 9.77614 7 9.5 7ZM11 7.5C11 7.22386 11.2239 7 11.5 7C11.7761 7 12 7.22386 12 7.5C12 7.77614 11.7761 8 11.5 8C11.2239 8 11 7.77614 11 7.5ZM11.5 9C11.2239 9 11 9.22386 11 9.5C11 9.77614 11.2239 10 11.5 10C11.7761 10 12 9.77614 12 9.5C12 9.22386 11.7761 9 11.5 9ZM9 9.5C9 9.22386 9.22386 9 9.5 9C9.77614 9 10 9.22386 10 9.5C10 9.77614 9.77614 10 9.5 10C9.22386 10 9 9.77614 9 9.5ZM7.5 9C7.22386 9 7 9.22386 7 9.5C7 9.77614 7.22386 10 7.5 10C7.77614 10 8 9.77614 8 9.5C8 9.22386 7.77614 9 7.5 9ZM5 9.5C5 9.22386 5.22386 9 5.5 9C5.77614 9 6 9.22386 6 9.5C6 9.77614 5.77614 10 5.5 10C5.22386 10 5 9.77614 5 9.5ZM3.5 9C3.22386 9 3 9.22386 3 9.5C3 9.77614 3.22386 10 3.5 10C3.77614 10 4 9.77614 4 9.5C4 9.22386 3.77614 9 3.5 9ZM3 11.5C3 11.2239 3.22386 11 3.5 11C3.77614 11 4 11.2239 4 11.5C4 11.7761 3.77614 12 3.5 12C3.22386 12 3 11.7761 3 11.5ZM5.5 11C5.22386 11 5 11.2239 5 11.5C5 11.7761 5.22386 12 5.5 12C5.77614 12 6 11.7761 6 11.5C6 11.2239 5.77614 11 5.5 11ZM7 11.5C7 11.2239 7.22386 11 7.5 11C7.77614 11 8 11.2239 8 11.5C8 11.7761 7.77614 12 7.5 12C7.22386 12 7 11.7761 7 11.5ZM9.5 11C9.22386 11 9 11.2239 9 11.5C9 11.7761 9.22386 12 9.5 12C9.77614 12 10 11.7761 10 11.5C10 11.2239 9.77614 11 9.5 11Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            Pick a date
          </span>
        </button>
      </div>
    </div>
    <button
      class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      type="submit"
    >
      Create Role
    </button>
  </form>
</DocumentFragment>
`;
